lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -480,6 +485,173 @@ def post(self, request):
             branch = request.POST.get('repo_branch', '').strip().replace(' ', '').replace(';', '')
             self.msg += self.get_course_from_git(gitloc, branch)
 
+        elif action == 'download_course_users':
+            course_id = request.POST.get('course_id', '').strip()
+            course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+            course_found = False
+            if course_key in courses:
+                course_found = True
+                course = courses[course_key]
+            else:
+                try:
+                    course = get_course_by_id(course_key)
+                    course_found = True
+                except Exception, err:   # pylint: disable=broad-except
+                    self.msg += _(
+                        'Error - cannot get course with ID {0}<br/><pre>{1}</pre>'
+                    ).format(
+                        course_key,
+                        escape(str(err))
+                    )
+            if course_found:
+                # donload csv list of cource students
+                data = []
+                # Remove current site orgs from the "filter out" list, if applicable.
+                # We want to filter and only show enrollments for courses within
+                # the organizations defined in configuration for the current site.
+                org_filter_out_set = configuration_helpers.get_all_orgs()
+                course_org_filter = configuration_helpers.get_current_site_orgs()
+                if course_org_filter:
+                    org_filter_out_set = org_filter_out_set - set(course_org_filter)
+
+                enrolled_students = User.objects.filter(
+                    courseenrollment__course_id=course_key)
+
+                # returns a list of chapters ids
+                # for parsing children (sequentials) in course tree
+                course_chapters = get_course_chapters(course_key)
+
+                course_obj = CourseStructure.objects.get(course_id=course_key)  # course content tree, serialised
+
+                # deserialising and converting to OrderedDict with the blocks
+                # in the order that they appear in the course
+                course_ordered = course_obj.ordered_blocks  # OrderedDict
+
+                def get_children(parent):
+                    """
+                    Method for getting children from course structure object
```

Comment:
please add the type of the parent arg into docstring and move this method definition into the current module